Backend/app/kafka-consumers/src/Consumer/LocationUpdate/Processor.hs

```diff
@@ -32,7 +33,25 @@ processLocationData T.LocationUpdates {..} driverId = do
   let newUpdatedAt = ts
   logDebug $ "driver updated time " <> show newUpdatedAt <> driverId
   let encodedVal = createDriverIdTokenKey driverId
-  Redis.zAdd "driver-last-location-update" [(utcToDouble newUpdatedAt, encodedVal)]
+  key <- incrementCounterAndReturnShard
+  Redis.zAdd key [(utcToDouble newUpdatedAt, encodedVal)]
+
+processLocationData' :: [(T.LocationUpdates, T.DriverId)] -> Flow ()
+processLocationData' locationData = do
+  key <- incrementCounterAndReturnShard
+  let encodedVals = map (\(T.LocationUpdates {..}, driverId) -> (utcToDouble ts, createDriverIdTokenKey driverId)) locationData
```

Comment:
This is actually not required driverIdTokenKey, can you remove this and use simple driverId.getId and in consumer side also make the relevant changes....